### PR TITLE
More Makefile cleanup/edits from old Makefile stuff

### PR DESCRIPTION
**Pull request by duglin**
Makefile edits
Lint fixes

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,66 +28,62 @@ endif
...
-$(BINDIR)/apiserver: cmd/service-catalog
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/apiserver: cmd/service-catalog $(shell find pkg/apis/servicecatalog -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/cmd/service-catalog
 
 # Some prereq stuff
 ###################
-init: .scBuildImage .init
-
-# .init is used to know when some Glide dependencies are out of date
```
Comment (MHBauer) [inline Makefile]: why remove these comments?

Comment (duglin) [inline Makefile]: because I got rid of the "init" target, its just ".init" now, so it obvious what its for based on the rule.

Comment (MHBauer) [inline Makefile]: not sure I agree with that.

Comment (MHBauer) [inline Makefile]: Alright, looks like that touch pattern is a known thing.
https://www.gnu.org/software/make/manual/html_node/Empty-Targets.html

Comment (MHBauer) [inline Makefile]: maybe a comments saying "empty targets for blah blah blah" ?

**Comment by MHBauer** [comment]
I'd prefer to have the lint commits separate from the build modifications.

LGTM

File: Makefile [hunk 2]
```diff
@@ -28,74 +27,78 @@ endif
...
 
 registry: $(BINDIR)/registry
-$(BINDIR)/registry: contrib/registry
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/registry: contrib/registry $(shell find contrib/registry -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/contrib/registry
 
-$(BINDIR)/k8s-broker: contrib/broker/k8s
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/k8s-broker: contrib/broker/k8s $(shell find contrib/broker/k8s -type f)
```
Comment (martinmaly) [inline Makefile]: this doesn't seem correct as it doesn't take into effect all dependencies:
https://github.com/kubernetes-incubator/service-catalog/blob/master/contrib/broker/k8s/broker.go#L26

I'm quite sure other rules have the same problem.

Comment (duglin) [inline Makefile]: is there a good way to get all dependencies for go files? Don't really want to create a specialized tool right now - so I figured this at least should get a fair amount of them.

Comment (martinmaly) [inline Makefile]: Personally I'd rather have false positives (we build when not needed) than false negatives (we don't build when we should). So I'd rather go for a `.PHONY` target and build always thank having inaccurate dependencies. But that's just me.

**Comment by arschles** [comment]
Looks like a good enough foundation to merge and iterate on in follow-ups

File: .gitignore [hunk 3]
```diff
@@ -13,3 +13,5 @@ coverage.html
 .glide
 .scBuildImage
 .init
+.generate*
+**/zz*.go
```
Comment (pmorie) [inline .gitignore]: This should not be in gitignore -- these files are checked in.

Comment (duglin) [inline .gitignore]: Why would we checked in generated files?

Comment (pmorie) [inline .gitignore]: We do it in k8s.

Comment (duglin) [inline .gitignore]: Was hoping for more of a reason than that :-)

Comment (pmorie) [inline .gitignore]: I think that is sufficient reason to check them in for now. I just woke up, and right now you get argument to authority. Please let's not try to be the tail that wags the dog on this yet. Going to be very hard to get help debugging if we don't do what upstream does.

**Comment by arschles** [comment]
Fwiw checking in generated code is done in more repos than just kubernetes/kubernetes. The helm project, among others that I know of, does it as well.

**Comment by duglin** [comment]
moved the "controversial" .gitignore stuff into the code-gen PR since its not really part of this clean-up work anyway.  We can continue the discussion over there.

File: Makefile (hunk 1 above)
Comment (pmorie) [inline Makefile]: I'd take those comments in a follow-up.

**Pull request merged**